include/boost/decimal/detail/emulated256.hpp

```diff
@@ -407,6 +378,58 @@ constexpr auto wide_integer_to_uint256(const wide_integer_uint256& src) -> uint2
     return dst;
 }
 
+constexpr uint256_t operator*(const uint256_t& lhs, const uint256_t& rhs) noexcept
+{
+    using local_unsigned_fast_type = ::boost::decimal::math::wide_integer::detail::unsigned_fast_type;
+
+    // Mash-Up: Use unrolled school-multiplication from wide-integer (requires limb-conversions on input/output).
+
+    auto lhs_wide = uint256_to_wide_integer(lhs);
+    auto rhs_wide = uint256_to_wide_integer(rhs);
+
+    wide_integer_uint256 result_wide { };
+
+    wide_integer_uint256::eval_multiply_n_by_n_to_lo_part(result_wide.representation().begin(),
```

Comment:
What's the type of `result_wide.representation()`? `std::array::begin` is not `constexpr` until C++17 and neither is `operator[]`.